### PR TITLE
Open XL header includes and vm flag fixup

### PR DESCRIPTION
**Pull request by Deigue**
runtime/tests/jsig/main.c : dlfcn.h needs to be defined before defines for dllfree()/dllload() due to conflicting type definitions decomp003.c/decomp005.c : strcasecmp() implict function declaration needs strings.h j9vmemTest.c : -Wimplicit-int , missing type specifier that defaults to "int"
jvminit.c : atoe_enableFileTagging() implicit function declaration requires atoe.h
runtime/vm/CMakeLists.txt : Added appropriate conditionals to apply correct flags for Open XL suggested by compiler team. Hooks are not implemented for Open XL.

### CODE REVIEW

File: runtime/vm/CMakeLists.txt
Comment (r30shah) [inline runtime/vm/CMakeLists.txt:58]: > runtime/vm/CMakeLists.txt : Added appropriate conditionals to apply correct flags for Open XL suggested by compiler team. Hooks are not implemented for Open XL.

Just to understand, with xlc it removes `-O3`, `-g` for compiler but with OpenXL it will not ?

**Review by r30shah — APPROVED** [review]
I just have one question I have asked, changes LGTM. Will launch test internally with this to make sure  we do not run into any issue before continuing to merge.

File: runtime/vm/CMakeLists.txt
Comment (joransiu) [reply runtime/vm/CMakeLists.txt:58]: Why won't we want `-O2` being added for `interp_new_flags` from the XLC case?

File: runtime/vm/jvminit.c
Comment (r30shah) [inline runtime/vm/jvminit.c:57]: @Deigue We may need this guarded by platform - See failures on other platform (http://vmfarm.rtp.raleigh.ibm.com/job_output.php?id=89250736)

**Review by r30shah — CHANGES_REQUESTED** [review]

File: runtime/vm/CMakeLists.txt
Comment (Deigue) [reply runtime/vm/CMakeLists.txt:58]: That might be an oversight, from the discussions I can find right now the intention was only to remove the `"\"-Wc` flags , but leave the same optimizations. So I will try with restoring the optimization levels and rebuilding.
If there is something I may not be remembering, at least it might remind me if it was intentional.

Comment (Deigue) [reply runtime/vm/CMakeLists.txt:58]: I think this is what the intended outcome was, which I am currently testing out ...

```
    if(OMR_TOOLCONFIG STREQUAL "openxl")                                                                                                                                                                                   # Hooks are not implemented in Open XL. The default debug compiles in Open XL will be NOHOOK
        # equivalent.
        list(APPEND interp_flags_to_remove "-O3" "-g")
        list(APPEND interp_new_flags -O2)
    else()
        # TODO need to handle optimized debug info
        list(APPEND interp_flags_to_remove "-O3" "-g" "-qdebug=nohook")

        list(APPEND interp_new_flags
            -O2
            "\"-Wc,TBYDBG(-qdebug=MRABIG)\""
            "\"-Wc,TBYDBG(-qdebug=lincomm:ptranl:tfbagg)\""
            "\"-Wc,FEDBG(-qxflag=InlineDespiteVolatileInArgs)\""
        )    
```

**Comment by Deigue** [comment]
I have removed `runtime/tests/jsig/main.c` reordering of headers as per @joransiu's advice. I discussed with Keith and have integrated this dependency where it should be (in the a2e wrappers, see `dll.h` change in https://github.com/eclipse/omr/pull/7383 , explicitly defined `dlfcn.h` as a pre-req for the functions it may override with its wrapper.

File: runtime/vm/jvminit.c
Comment (Deigue) [reply runtime/vm/jvminit.c:58]: Done

File: runtime/vm/CMakeLists.txt
Comment (Deigue) [reply runtime/vm/CMakeLists.txt]: Verified these changes are working with openxl, still need to test xlc with jenkins, but I imagine it won't be a problem since that clause/conditional has remained unchanged altogether.

**Comment by Deigue** [comment]
[jdk21 build](https://hyc-runtimes-jenkins.swg-devops.com/job/Build_JDK21_s390x_zos_Personal/1272/console) passes , same commit based off `ibm-openxl-headers` (internal fork)

**Comment by Deigue** [comment]
This one is ready @r30shah , and the dependent changes from https://github.com/eclipse/omr/pull/7383 fixing the headers has been merged. Can we get this reviewed/merged

**Comment by Deigue** [comment]
java 8 builds seems to be good, except for `Compose SE8 Release: xz6480` Job, but it doesn't seem directly related to the changes here.

**Comment by Deigue** [comment]
All passed with java 8 and java 21 builds.

**Review by joransiu — APPROVED** [review]

**Comment by r30shah** [comment]
jenkins test sanity zlinux jdk21

**Comment by r30shah** [comment]
Failing test in the launched test is same as https://github.com/eclipse-openj9/openj9/issues/19618, not related to change in this PR. Merging this as internal tests have been confirmed by @Deigue  and changes have been approved.

File: runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/decompResolveFrame/decomp003.c
Comment (r30shah) [inline runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/decompResolveFrame/decomp003.c:25]: @Deigue  Given that this is causing failure - I wonder, we should guard this with `J9ZOS` same as how `atoe.h` was done (Assuming that only z/OS needs the function defined in `strings.h`which you may have bumped into with OpenXL work.

Comment (Deigue) [reply runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/decompResolveFrame/decomp003.c:25]: yes that sounds like a good idea, I think the build must have failed on one of the platforms we didnt verify with.

**Pull request merged**
